src/bindable-property.js

```diff
@@ -58,17 +64,55 @@ export class BindableProperty {
   * @param descriptor The property descriptor for this property.
   */
   registerWith(target: Function, behavior: HtmlBehaviorResource, descriptor?: Object): void {
+    let { reflect } = this;
     behavior.properties.push(this);
     behavior.attributes[this.attribute] = this;
     this.owner = behavior;
 
+    if (reflect) {
+      behavior._registerReflection(this.name, typeof reflect === 'function' ? reflect : propToAttr);
+      this._configureReflection(target);
+    }
+
     if (descriptor) {
       this.descriptor = descriptor;
       return this._configureDescriptor(descriptor);
     }
 
     return undefined;
   }
+  
+  _configureReflection(target) {
+    if (target.__reflectionConfigured__) return;
+
+    let method = 'propertyChanged';
+    let onChanged = target.prototype[method];
+    let hasHandler = !!onChanged;
+    
+    let alteredHandler;
+    if (hasHandler) { // avoid ternary to make it consisten with the rest ?
+      alteredHandler = function propertyChanged(name, newVal, oldVal) {
+        onChanged.call(this, name, newVal, oldVal);
+        let { __element__, __reflections__ } = this.__observers__;
```

Comment:
Not sure, so just asking: is accessing `__observers__` directly the way to go?
Isn't that an implementation detail of observation and an API should be used here rather?

Reply:
This one is really silly of mine, was too hasty to see this. 👍 
How would you fix this though, there is already `getObserver` in the scope, in which it access `__observers__` directly. Maybe i should create `getObserverLookup` and point `getObserver` there. Or just leave `getObserver` alone to avoid performance cost of fn calling ?

Reply:
I did not see that `getObserver` function. I think the right thing to do is to use it to not break the abstraction level here.

If you want to maximize perfs and make sure `getObserver` gets optimized by JIT you can split it in two so that it is small and only contains the happy path (`lookup !== undefined`) and put the initialization code in a second function.

Reply:
I think I will move the whole block of reflection out of that function, to a separate fn